src/Features/UserImpersonation.php

```
@@ -76,16 +77,24 @@ public static function modelClass(): string
 
     public static function isImpersonating(): bool
     {
-        return session()->has('tenancy_impersonating');
+        return session()->has('tenancy_impersonation_guard');
     }
 
     /**
      * Logout from the current domain and forget impersonation session.
      */
-    public static function stopImpersonating(): void
+    public static function stopImpersonating(bool $logout = true): void
```

PHPMD (2.15.0)

[error] The method stopImpersonating has a boolean flag argument $logout, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)